addons/sale/models/sale_order.py

```diff
@@ -849,31 +847,44 @@ def _show_cancel_wizard(self):
         return False
 
     def _find_mail_template(self, force_confirmation_template=False):
-        template_id = False
+        """ Get the appropriate mail template for the current sale's order(s) based on their state.
 
-        if force_confirmation_template or (self.state == 'sale' and not self.env.context.get('proforma', False)):
-            template_id = int(self.env['ir.config_parameter'].sudo().get_param('sale.default_confirmation_template'))
-            template_id = self.env['mail.template'].search([('id', '=', template_id)]).id
-            if not template_id:
-                template_id = self.env['ir.model.data'].xmlid_to_res_id('sale.mail_template_sale_confirmation', raise_if_not_found=False)
-        if not template_id:
-            template_id = self.env['ir.model.data'].xmlid_to_res_id('sale.email_template_edi_sale', raise_if_not_found=False)
+        If all SOs are 'sale' or ``force_confirmation_template`` is `True`, the mail template is
+        that of the sale confirmation. Else, it returns the quotation's email template.
 
-        return template_id
+        :param bool force_confirmation_template: Allows to force the confirmation template to be
+        used in any case
+        :return: The correct mail template based on the current status
+        :rtype: record of `mail.template` or `None` if not found
+        """
+        if force_confirmation_template or (
+            all(order.state == 'sale' for order in self) and not self.env.context.get('proforma')
+        ):
+            default_confirmation_template_id = int(
+                self.env['ir.config_parameter']
+                    .sudo()
+                    .get_param('sale.default_confirmation_template')
+            )
+            default_confirmation_template = self.env['mail.template'].browse(
+                default_confirmation_template_id
+            )
+            return default_confirmation_template or self.env.ref(
+                'sale.mail_template_sale_confirmation', False
+            )
+        return self.env.ref('sale.email_template_edi_sale', False)
 
     def action_quotation_send(self):
         ''' Opens a wizard to compose an email, with relevant mail template loaded by default '''
         self.ensure_one()
-        template_id = self._find_mail_template()
+        mail_template = self._find_mail_template()
         lang = self.env.context.get('lang')
-        template = self.env['mail.template'].browse(template_id)
-        if template.lang:
-            lang = template._render_lang(self.ids)[self.id]
+        if mail_template.lang:
+            lang = mail_template._render_lang(self.ids)[self.id]
```

Comment:
Modified behavior:
Previously, `_find_mail_template` returned an id (or False if not found), then browsing it gave an empty recordset (which doesn't raise if you access its fields)

Now, it returns a recordset or None if not found, which will raise if the records are not found.

Or you should still be able to open the "send mail" wizard even if all the templates have been deleted.